File: OpenGLC.MVC/ClientApp/src/app/Models/NewMealEventModel.ts
export interface NewMealEventModel {
  eventDate: Date;
  glcLevel: number;
  newMeals: NewMealItemPair[];
  itemMeals: ExistingMealItemPair[];
  postprandial: boolean;
  mealType: number;
}

export interface NewMealItemPair {
  name: string;
  quantity: number;
}

export interface ExistingMealItemPair {
  iD: string;
  quantity: number;
}

export interface NewMealItemModelDB {
  id: string;
  name: string;
  quantity: number;
}

File: OpenGLC.Models/MealEvents/NewMealEventModel.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace OpenGLC.Models.MealEvents
{
	public class NewMealEventModel
	{
		public NewMealEventModel()
		{
			NewMeals = new List<NewMealItemPair>();
			ItemMeals = new List<ExistingMealItemPair>();
		}
		public DateTime EventDate { get; set; }
		public int GlcLevel { get; set; }
		public List<NewMealItemPair> NewMeals { get; set; }
		public List<ExistingMealItemPair> ItemMeals { get; set; }
		public bool Postprandial { get; set; }
		public int MealType { get; set; }
	}

	public class NewMealItemPair
	{
		public string Name { get; set; }
		public int Quantity { get; set; }
	}

	public class ExistingMealItemPair
	{
		public Guid ID { get; set; }
		public int Quantity { get; set; }
	}
}
